AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift

```diff
@@ -178,65 +196,82 @@ class PatrolAutomation {
     }
   }
   
+  // MARK: Notifications
+  
+  func openNotifications() {
+    // TODO: Check if works on iPhones without notch
+    
+    runAction("opening notifications") {
+      let start = self.springboard.coordinate(withNormalizedOffset: CGVector(dx: 0.5, dy: 0.01))
+      let end = self.springboard.coordinate(withNormalizedOffset: CGVector(dx: 0.5, dy: 0.6))
+      start.press(forDuration: 0.1, thenDragTo: end)
+    }
+  }
+  
+  func closeNotifications() {
+    // TODO: Check if works on iPhones without notch
+    
+    runAction("closing notifications") {
+      let start = self.springboard.coordinate(withNormalizedOffset: CGVector(dx: 0.5, dy: 0.99))
+      let end = self.springboard.coordinate(withNormalizedOffset: CGVector(dx: 0.5, dy: 0.6))
+      start.press(forDuration: 0.1, thenDragTo: end)
+    }
+  }
+  
   func getNotifications() -> [Patrol_Notification] {
-    let notifications = [Patrol_Notification]()
+    var notifications = [Patrol_Notification]()
     runAction("getting notifications") {
-      let cells = self.springboard.buttons.allElementsBoundByIndex
-      for cell in cells {
+      let cells = self.springboard.buttons.matching(identifier: "NotificationCell").allElementsBoundByIndex
+      for (i, cell) in cells.enumerated() {
         let notification = Patrol_Notification.with {
-          $0.appName = cell.label.components(separatedBy: ",")[0]
-          $0.content = cell.label.components(separatedBy: ",")[1]
+          Logger.shared.i("found notification at index \(i) with label \(format: cell.label)")
+          $0.raw = cell.label
         }
+        notifications.append(notification)
       }
     }
     
     return notifications
   }
-
-  func tap(on text: String, inApp appId: String) {
-    runAction("tapping on \(text)") {
-      let app = XCUIApplication(bundleIdentifier: appId)
-      app.descendants(matching: .any)[text].firstMatch.tap()
-    }
-  }
-
-  func doubleTap(on text: String, inApp appId: String) {
-    runAction("double tapping on \"\(text)\"") {
-      let app = XCUIApplication(bundleIdentifier: appId)
-      app.descendants(matching: .any)[text].firstMatch.tap()
-    }
-  }
-
-  func enterText(_ data: String, by text: String, inApp appId: String) {
-    runAction("entering text \"\(text)\"") {
-      let app = XCUIApplication(bundleIdentifier: appId)
-      app.textFields[text].firstMatch.typeText(data)
+  
+  func tapOnNotification(by index: Int) {
+    runAction("tapping on notification at index \(index)") {
+      let cells = self.springboard.buttons.matching(identifier: "NotificationCell").allElementsBoundByIndex
+      if !cells.indices.contains(index) {
+        Logger.shared.e("notification at index \(index) doesn't exist")
+      }
+      
+      cells[index].tap()
```

Comment:
Do you still want to tap if it does not exist?

Reply:
Ah, a `return` should be there.